bundles/org.openhab.binding.alarmdecoder/src/main/java/org/openhab/binding/alarmdecoder/internal/protocol/KeypadMessage.java

```diff
@@ -56,7 +56,8 @@
 
     public KeypadMessage(String message) throws IllegalArgumentException {
         super(message);
-        List<String> parts = splitMsg(message);
+        String topLevel[] = message.split(":");
```

Comment:
I can't say for sure that the keypad message string will never include a ":" character, so doing a split on ":" could potentially cause a problem at some point. It may be better to use something like ```message.replace("!KPM:", "")```.

Reply:
Sure thing, will change!!

Reply:
@bobadair Done!